jwala-webservices/src/main/java/com/cerner/jwala/ws/rest/v1/service/group/impl/GroupServiceRestImpl.java

```diff
@@ -759,29 +798,44 @@ protected void performGroupAppDeployToJvms(final String groupName, final String
         } else {
             LOGGER.debug("got no hostname deploying to all group jvms");
             jvms = groupJvms;
+        } return jvms;
+    }
+
+    private Map<String, Future<Response>> executeGroupAppDeployToJvms(final String groupName, final String fileName, final AuthenticatedUser aUser, final String appName, final ApplicationServiceRest appServiceRest, Set<Jvm> jvms) {
+        final String groupAppTemplateContent = groupService.getGroupAppResourceTemplate(groupName, appName, fileName, false, new ResourceGroup());
+        final Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+        final  Map<String, Future<Response>> futureMap = new HashMap<>();
+        for (Jvm jvm : jvms) {
+            final String jvmName = jvm.getJvmName();
+            Future<Response> responseFuture = executorService.submit(new Callable<Response>() {
+                @Override
+                public Response call() throws Exception {
+                    SecurityContextHolder.getContext().setAuthentication(auth);
+                    appServiceRest.updateResourceTemplate(appName, fileName, jvmName, groupName, groupAppTemplateContent);
+                    return appServiceRest.deployConf(appName, groupName, jvmName, fileName, aUser);
+                }
+            });
+            futureMap.put(jvmName, responseFuture);
         }
-        if (null != jvms && !jvms.isEmpty()) {
-            for (Jvm jvm : jvms) {
-                if (jvm.getState().isStartedState()) {
-                    LOGGER.info("Failed to deploy file {} for group {}: not all JVMs were stopped - {} was started", fileName, group.getName(), jvm.getJvmName());
-                    throw new InternalErrorException(FaultType.REMOTE_COMMAND_FAILURE, "All JVMs in the group must be stopped before continuing. Operation stopped for JVM " + jvm.getJvmName());
+        return futureMap;
+    }
+
+    private void checkJvmsStatesBeforeDeployAppResource(String fileName, Group group, boolean hotDeploy, Set<Jvm> jvms) {
+        List<String> jvmsStarted = new ArrayList<>();
+        for (Jvm jvm : jvms) {
+            if (jvm.getState().isStartedState()) {
+                if (hotDeploy) {
```

Comment:
Can we change to

    if (started && !hotDeploy) {
        ...
        continue;
    }
    LOGGER.info(...)